### PR TITLE
Add criteria for supported providers

### PR DESCRIPTION
**Pull request by ritazh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -240,25 +240,22 @@ csi-secrets-store-provider-azure-sxht2             2/2     Running   0
 
 ## Providers
 
-This project features a pluggable provider interface developers can implement that defines the actions of the Secrets Store CSI driver.
+This project features a pluggable provider interface developers can implement that defines the actions of the Secrets Store CSI driver. This enables retrieval of sensitive objects storied an enterprise-grade external secrets store into Kubernetes while continue to manage these objects outside of Kubernetes.
```
Comment (aramase) [inline README.md]: Should it be `This enables retrieval of sensitive objects stored in enterprise-grade external secret stores into Kubernetes`? Do we need to mention `managed these objects outside of kubernetes`?

File: README.md [hunk 2]
```diff
@@ -240,25 +240,22 @@ csi-secrets-store-provider-azure-sxht2             2/2     Running   0
...
-1. Does not have access to the Kubernetes APIs and has a well-defined callback mechanism to mount objects to a target path.
-
-- Supported Providers:
-  - [Azure Key Vault Provider](https://github.com/Azure/secrets-store-csi-driver-provider-azure)
-  - [HashiCorp Vault Provider](https://github.com/hashicorp/secrets-store-csi-driver-provider-vault)
-
-### Adding a New Provider via the Provider Interface
-
-_WIP_
+when a provider's e2e tests are consistently failing with the latest version of the driver, the driver maintainers will coordinate with the provider maintainers to provide a fix. If the test failures prolong to more than 4 weeks, then the provider will be removed from the list of supported providers.
```
Comment (aramase) [inline README.md]: nit: when -> `When`

Comment (aramase) [reply README.md]: Should we also say failure to adhere to the criteria will also result in provider being removed from the list of supported through the lifecycle of the project?

File: README.md (hunk 1 above)
Comment (craiglpeters) [reply README.md]: I think it is useful to contrast this approach with Kubernetes secrets in this way

File: README.md [hunk 3]
```diff
@@ -240,25 +240,22 @@ csi-secrets-store-provider-azure-sxht2             2/2     Running   0
...
 ## Providers
 
-This project features a pluggable provider interface developers can implement that defines the actions of the Secrets Store CSI driver.
+This project features a pluggable provider interface developers can implement that defines the actions of the Secrets Store CSI driver. This enables retrieval of sensitive objects storied an enterprise-grade external secrets store into Kubernetes while continue to manage these objects outside of Kubernetes.
 
-This enables on-demand retrieval of sensitive objects storied an enterprise-grade external secrets store into Kubernetes while continue to manage these objects outside of Kubernetes.
+### Criteria for Supported Providers
 
-Each provider may have its own required properties.
+Here is a list of criteria for being added as a supported provider:
```
Comment (craiglpeters) [inline README.md]: Criteria for providers:

File: README.md [hunk 4]
```diff
@@ -240,25 +240,22 @@ csi-secrets-store-provider-azure-sxht2             2/2     Running   0
...
+1. Code audit of the provider implementation to ensure it adheres to the required provider-driver interface, which includes:
+    - implementation of provider command args https://github.com/deislabs/secrets-store-csi-driver/blob/master/pkg/secrets-store/nodeserver.go#L223-L236
+    - provider binary naming convention and semver convention
+    - provider binary deployment volume path
+    - provider logs are written to stdout and stderr so they can be part of the driver logs
+1. Add provider to the e2e test suite to demonstrate it functions as expected https://github.com/deislabs/secrets-store-csi-driver/tree/master/test/bats
```
Comment (craiglpeters) [inline README.md]: Is the "how" for this documented in the tests, or somewhere else?

File: README.md [hunk 5]
```diff
@@ -240,25 +240,22 @@ csi-secrets-store-provider-azure-sxht2             2/2     Running   0
...
+1. If any update is made by a provider (not limited to security updates), the provider is expected to update the provider's e2e test in this repo
 
-Providers must provide the following functionality to be considered a supported integration.
+### Removal from Supported Providers
 
-1. Provides the backend plumbing necessary to access objects from the external secrets store.
-1. Conforms to the current API provided by the Secrets Store CSI Driver.
-1. Does not have access to the Kubernetes APIs and has a well-defined callback mechanism to mount objects to a target path.
-
-- Supported Providers:
```
Comment (craiglpeters) [inline README.md]: With this removal, how will users learn about new providers?

Comment (ritazh) [reply README.md]: The list of providers are linked at the very top of the readme.

File: README.md (hunk 4 above)
Comment (ritazh) [reply README.md]: Will update to `Add provider to the e2e test suite to demonstrate it functions as expected https://github.com/deislabs/secrets-store-csi-driver/tree/master/test/bats Please use existing providers e2e tests as a reference.`

File: README.md (hunk 1 above)
Comment (aramase) [reply README.md]: nit: `storied in` to `stored in`

**Pull request merged**